scalyr_agent/configuration.py

```diff
@@ -1035,6 +1093,80 @@ def __verify_main_config_and_apply_defaults(self, config, file_path, apply_defau
         self.__verify_or_set_optional_int(config, 'disable_leak_verify_config_create_copying_manager', None, description, apply_defaults)
         self.__verify_or_set_optional_bool(config, 'disable_leak_verify_config_cache_config', False, description, apply_defaults)
 
+    def __get_config_or_environment_val(self, config_object, param_name, param_type):
```

Comment:
Encapsulate getting of config file value with getting of environment value as well.
Also add support for newly defined `ArrayOfStrings` type (for type safety)